### PR TITLE
[bitnami/appsmith] Add VIB tests

### PR DESCRIPTION
**Pull request by FraPazGal**
Signed-off-by: FraPazGal <fdepaz@vmware.com>

### Description of the change

The main objective of this PR is to publish our Bitnami Appsmith container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD. 

### Applicable issues

NA

Tested in:
- Recent action run: https://github.com/bitnami/containers/actions/runs/4745248986

### CODE REVIEW

File: .vib/common/goss/templates/check-linked-libraries.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ command:
         {{ $exclude_paths = (.Vars.linked_libraries.exclude_paths | join "|") }}
     {{ end }}
     exec: export BITNAMI_ROOT_DIR={{ .Vars.root_dir }} && export EXCLUDE_PATHS='{{ $exclude_paths }}' && ./common/goss/scripts/check-linked-libraries.sh
-    timeout: 20000
```
Comment (migruiz4) [inline .vib/common/goss/templates/check-linked-libraries.yaml:15]: @FraPazGal @joancafom Should we increase the timeout for all assets? Or should we instead add support for a customizable timeout?

In [this PR](https://github.com/bitnami/containers/pull/30858/files#diff-482f3343f255b156bae488ed2c775287ae0d7e03bc6b637c046234ea561f4212) I implemented the option to include `linked_libraries.timeout` in the vars.yaml file for those cases were a bigger timeout is required.

What do you think?

File: .vib/appsmith/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - /opt/bitnami/appsmith/conf
+      - /opt/bitnami/appsmith/logs
+      - /opt/bitnami/appsmith/tmp
+  - paths:
+      - /opt/bitnami/appsmith/backend/plugins
+      - /opt/bitnami/appsmith/editor
+      - /opt/bitnami/appsmith/templates/client
+files:
+  - paths:
+      - /opt/bitnami/appsmith/conf/docker.env
```
Comment (migruiz4) [inline .vib/appsmith/goss/vars.yaml:22]: Shouldn't we add `mode` check for `/opt/bitnami/appsmith/conf/docker.env` considering they are explicitly set in the Appsmith postunpack logic?

File: .vib/appsmith/goss/vars.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+binaries:
+  - java
+  - mongosh
+  - nginx
+  - node
+  - python
+  - render-template
+  - wait-for-port
+directories:
```
Comment (migruiz4) [inline .vib/appsmith/goss/vars.yaml:9]: Shouldn't we also check for files and directories added by the `appsmith` package?

E.g. `backend/server.jar`, `editor`, `rts/package.json`, `templates`, `templates/client`.

Comment (FraPazGal) [reply .vib/appsmith/goss/vars.yaml:9]: There were a couple of files I missed checking, added those now. Thanks!

**Comment by migruiz4** [comment]
Could you please resolve the conflicts and implement `linked_libraries.timeout` as discussed here https://github.com/bitnami/containers/pull/30871/files#r1172209216?

**Review by agomezmoron — DISMISSED** [review]

**Review by migruiz4 — APPROVED** [review]
LGTM!

**Pull request merged**